pkg/apis/serving/metadata_validation.go

```diff
@@ -76,3 +76,12 @@ func ValidateTimeoutSeconds(ctx context.Context, timeoutSeconds int64) *apis.Fie
 	}
 	return nil
 }
+
+// ValidateContainerConcurrency function validates the ContainerConcurrency field
+func ValidateContainerConcurrency(ctx context.Context, containerConcurrency int64) *apis.FieldError {
```

Comment:
I don't think I would have put this here (it's not actually metadata), but I'm planning to move it as part of https://github.com/knative/serving/pull/5007 anyway, so it's harmless.

Maybe add a `TODO(#5007): Move this to autoscaling.`?